packages/gatsby/src/utils/hashed-chunk-ids-plugin.js

```diff
@@ -14,7 +14,7 @@ HashedChunkIdsPlugin.prototype.apply = function apply(compiler) {
   compiler.plugin(`compilation`, compilation => {
     compilation.plugin(`before-chunk-ids`, chunks => {
       chunks.forEach(chunk => {
-        if (chunk.id === null) {
+        if (chunk.id === null && chunk.name != null) {
```

Comment:
Now that we're on webpack 2 we can use NamedModuleIds and NamedChunkIds per this article https://medium.com/webpack/predictable-long-term-caching-with-webpack-d3eee1d3fa31

This plugin was added recently to work around webpack 1 deficiencies so now can be removed.